client/lib/user/user.js

```diff
@@ -260,7 +260,7 @@ User.prototype.clear = function( onClear ) {
 	delete this.settings;
 	store.clearAll();
 	if ( config.isEnabled( 'persist-redux' ) ) {
-		localforage.clear( onClear );
+		clearStorage().then( onClear );
 	} else if ( onClear ) {
 		onClear();
```

Comment:
maybe this should be `Promise.resolve().then( onClear )` to prevent clear from running sync in some cases?

Reply:
I'm not sure I follow. Did you mean `Promise.resolve().then(() => clearStorage()).then(onClear)` ? And if so, is the goal to ensure that there's always a microtask wait between this being called and the `then` being called?

If so, I'd argue that we should instead ensure that all executions are behind a microtask wait in the lib implementation, rather than the consumers.

Reply:
no, I'm actually calling out the line not currently in the diff. I just noticed it on a drive-by. 

If persist-redux is enabled, `onClear` is called via promise resolution. If it is _not_ enabled, we call it synchronously. I think we should call it on the next microtask to prevent the call timing from changing.

Reply:
Oh, sorry, totally misread you there. Sounds like a reasonable change :+1:

Reply:
Done.